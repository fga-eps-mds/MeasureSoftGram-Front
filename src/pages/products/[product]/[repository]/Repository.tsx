import React, { useEffect, useState } from 'react';
import { Box, Container, Typography } from '@mui/material';

import Filters from '@components/Filters';
import getLayout from '@components/Layout';
import GraphicStackedLine from '@components/GraphicStackedLine';

import { NextPageWithLayout } from '@pages/_app.next';

import { useRepositoryContext } from '@contexts/RepositoryProvider';

import Skeleton from './components/Skeleton';
import SubCharacteristicsList from './components/SubCharacteristicsList';

import { useQuery as useQueryProduct } from '../hooks/useQuery';
import { useQuery } from './hooks/useQuery';

import * as Styles from './styles';

interface FilterProps {
  filterTitle: string;
  options: Array<string>;
}

const Repository: NextPageWithLayout = () => {
  useQueryProduct();

  const { repositoryHistoricalCharacteristics, checkedOptionsFormat } = useQuery();
  const { characteristics, subCharacteristics, currentRepository, historicalSQC } = useRepositoryContext();

  const [filterCharacteristics, setFilterCharacteristics] = useState<FilterProps>({
    filterTitle: 'CARACTERÍSTICAS',
    options: []
  });
  const [filterSubCharacteristics, setFilterSubCharacteristics] = useState<FilterProps>({
    filterTitle: 'SUB CARACTERÍSTICAS',
    options: []
  });

  const [checkedOptions, setCheckedOptions] = useState(checkedOptionsFormat);

  useEffect(() => {
    setCheckedOptions(checkedOptionsFormat);
  }, [checkedOptionsFormat]);

  useEffect(() => {
    setFilterCharacteristics({ ...filterCharacteristics, options: characteristics });
    setFilterSubCharacteristics({ ...filterSubCharacteristics, options: subCharacteristics });
  }, [characteristics, subCharacteristics]);

  const isArrayEmpty = (array: Array<any>) => array.length === 0;

  if (
    isArrayEmpty(repositoryHistoricalCharacteristics) ||
    isArrayEmpty(filterCharacteristics.options) ||
    isArrayEmpty(filterSubCharacteristics.options) ||
    !currentRepository ||
    !historicalSQC
  ) {
    return <Skeleton />;
  }

  return (
    <Box display="flex" width="100%" flexDirection="row">
      <Styles.FilterBackground>
        <Box display="flex" paddingX="15px" flexDirection="column" marginTop="36px" position="fixed">
          {[filterCharacteristics, filterSubCharacteristics].map((filter) => (
            <Filters
              key={filter.filterTitle}
              filterTitle={filter.filterTitle}
              options={filter.options}
              updateOptions={setCheckedOptions}
              checkedOptions={checkedOptions}
            />
          ))}
        </Box>
      </Styles.FilterBackground>

      <Box width="100%">
        <Box marginBottom="42px">
          <Container>
            <Box display="flex" flexDirection="column" width="100%">
              <Box marginTop="40px" marginBottom="36px">
                <Box display="flex">
                  <Typography variant="h4" marginRight="10px">
                    Repositório
                  </Typography>
                  <Typography variant="h4" fontWeight="300">
                    {currentRepository?.name}
                  </Typography>
                </Box>

                <Typography variant="caption" color="gray">
                  {currentRepository?.description}
                </Typography>
              </Box>

              {historicalSQC &&
                repositoryHistoricalCharacteristics &&
                repositoryHistoricalCharacteristics.length !== 0 && (
                  <GraphicStackedLine
                    historical={repositoryHistoricalCharacteristics.concat(historicalSQC)}
                    checkedOptions={checkedOptions}
                    title="Características"
                  />
                )}
            </Box>
          </Container>
        </Box>

        <SubCharacteristicsList checkedOptions={checkedOptions} />
      </Box>
    </Box>
  );
};

Repository.getLayout = getLayout;

export default Repository;
