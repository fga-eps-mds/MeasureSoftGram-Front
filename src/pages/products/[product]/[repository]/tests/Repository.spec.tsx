import '@testing-library/jest-dom';

import React from 'react';
import { render } from '@testing-library/react';

import Repository from '../Repository';

jest.mock('@contexts/ProductProvider', () => ({
  useProductContext: () => ({
    currentProduct: {
      name: `aoba`
    }
  })
}));

jest.mock('@contexts/RepositoryProvider', () => ({
  useRepositoryContext: () => ({})
}));

describe('Repository', () => {
  describe('Snapshot', () => {
    it('Deve corresponder ao Snapshot', () => {
      const tree = render(<Repository />);

      expect(tree).toMatchSnapshot();
    });
  });
});
