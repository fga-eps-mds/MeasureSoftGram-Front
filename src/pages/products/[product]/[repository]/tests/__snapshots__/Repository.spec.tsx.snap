// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository Snapshot Deve corresponder ao Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-kzcie9"
      >
        <div
          class="styles__FilterBackground-sc-n9f0ab-0 gJDcCB"
        >
          <div
            class="MuiBox-root css-11dhwbz"
          >
            <div>
              <button
                class="styles__FiltersTitle-sc-1ywle5j-0 ePgPyn"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ExpandCircleDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11 12 15.5z"
                  />
                </svg>
                <span>
                  CARACTERÍSTICAS
                </span>
              </button>
              <div
                class="styles__OptionContainer-sc-1ywle5j-1 MuiBox-root css-j7qwjs"
              />
            </div>
            <div>
              <button
                class="styles__FiltersTitle-sc-1ywle5j-0 ePgPyn"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ExpandCircleDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11 12 15.5z"
                  />
                </svg>
                <span>
                  SUB CARACTERÍSTICAS
                </span>
              </button>
              <div
                class="styles__OptionContainer-sc-1ywle5j-1 MuiBox-root css-j7qwjs"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <div
            class="MuiBox-root css-1ofqig9"
          >
            <div
              class="MuiBox-root css-10bo7nf"
            >
              <div
                class="MuiBox-root css-k008qs"
              >
                <h4
                  class="MuiTypography-root MuiTypography-h4 css-obdthj-MuiTypography-root"
                >
                  Repositório
                </h4>
                <h4
                  class="MuiTypography-root MuiTypography-h4 css-1yvpk5c-MuiTypography-root"
                />
              </div>
              <span
                class="MuiTypography-root MuiTypography-caption css-qkcnm0-MuiTypography-root"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-kzcie9"
    >
      <div
        class="styles__FilterBackground-sc-n9f0ab-0 gJDcCB"
      >
        <div
          class="MuiBox-root css-11dhwbz"
        >
          <div>
            <button
              class="styles__FiltersTitle-sc-1ywle5j-0 ePgPyn"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ExpandCircleDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11 12 15.5z"
                />
              </svg>
              <span>
                CARACTERÍSTICAS
              </span>
            </button>
            <div
              class="styles__OptionContainer-sc-1ywle5j-1 MuiBox-root css-j7qwjs"
            />
          </div>
          <div>
            <button
              class="styles__FiltersTitle-sc-1ywle5j-0 ePgPyn"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ExpandCircleDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 13.5L7.5 11l1.42-1.41L12 12.67l3.08-3.08L16.5 11 12 15.5z"
                />
              </svg>
              <span>
                SUB CARACTERÍSTICAS
              </span>
            </button>
            <div
              class="styles__OptionContainer-sc-1ywle5j-1 MuiBox-root css-j7qwjs"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-1ofqig9"
        >
          <div
            class="MuiBox-root css-10bo7nf"
          >
            <div
              class="MuiBox-root css-k008qs"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4 css-obdthj-MuiTypography-root"
              >
                Repositório
              </h4>
              <h4
                class="MuiTypography-root MuiTypography-h4 css-1yvpk5c-MuiTypography-root"
              />
            </div>
            <span
              class="MuiTypography-root MuiTypography-caption css-qkcnm0-MuiTypography-root"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
